packages/components/src/components/hds/table/index.ts

```diff
@@ -92,11 +95,48 @@ export default class HdsTableIndexComponent extends Component<HdsTableArgs> {
   selectableRows: HdsTableSelectableRow[] = [];
   @tracked isSelectAllCheckboxSelected?: boolean = undefined;
 
+  sortModel = (
+    sortCriteria: string | HdsTableSortingFunction<unknown>,
+    model: typeof this.args.model
+  ): typeof this.args.model => {
+    if (sortCriteria === '__selected__') {
+      const { selectedModels, unselectedModels } = this.args.model.reduce(
+        (
+          acc: { selectedModels: unknown[]; unselectedModels: unknown[] },
+          model
+        ) => {
+          if (model[this.args.selectedItemKey]) {
+            acc.selectedModels.push(model);
+          } else {
+            acc.unselectedModels.push(model);
+          }
+          return acc;
+        },
+        { selectedModels: [], unselectedModels: [] }
+      );
+
+      console.log(selectedModels, unselectedModels);
+
+      return [...selectedModels, ...unselectedModels] as typeof this.args.model;
+    }
+
+    if (typeof sortCriteria === 'string') {
+      return A([...model]).sortBy(sortCriteria);
```

Comment:
even though direct usage of ember array is allowed it is discouraged, additionally usage of ember array prototype extensions is deprecated and will be removed. https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-array-prototype-extensions.md

Disregard this comment if this is only done do demo but I thought I would call this out

Reply:
Thanks for calling that out, but yeah, this is just a little hacky demo.

Reply:
Nonetheless, `eslint-plugin-ember` does need updating. Will make a note of that